statsd/src/main/java/com/avast/metrics/statsd/StatsDTimer.java

```diff
@@ -0,0 +1,161 @@
+package com.avast.metrics.statsd;
+
+import com.avast.metrics.api.Timer;
+import com.timgroup.statsd.StatsDClient;
+
+import java.time.Duration;
+import java.util.Optional;
+import java.util.concurrent.Callable;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+
+@SuppressWarnings("Duplicates")
+public class StatsDTimer implements Timer {
+    private final StatsDClient client;
+    private final String name;
+
+    private final AtomicLong count = new AtomicLong(0);
+
+    StatsDTimer(final StatsDClient client, final String name) {
+        this.client = client;
+        this.name = name;
+    }
+
+    @Override
+    public String getName() {
+        return name;
+    }
+
+    @Override
+    public TimeContext start() {
+        return new StatsDTimerContext(this::update);
+    }
+
+    @Override
+    public void update(final Duration duration) {
+        count.incrementAndGet();
+        client.recordExecutionTime(name, duration.toMillis());
+    }
+
+    @Override
+    public <T> T time(final Callable<T> operation) throws Exception {
+        final TimeContext context = start();
+        final T result = operation.call();
+        context.stop();
+        return result;
+    }
+
+    @Override
+    public <T> T time(final Callable<T> operation, final Timer failureTimer) throws Exception {
+        final TimeContext successContext = start();
+        final TimeContext failureContext = failureTimer.start();
+
+        try {
+            final T result = operation.call();
+            successContext.stop();
+            return result;
+        } catch (Exception e) {
+            failureContext.stop();
+            throw e;
+        }
+    }
+
+    @Override
+    public <T> CompletableFuture<T> timeAsync(final Callable<CompletableFuture<T>> operation, final Executor executor) throws Exception {
```

Comment:
I am wondering, do these methods have to be reimplemented? Can't you just use the already implemented metrics in core and just record the result in StatsD?